### ISSUE TITLE
Airflow 1.10.10 blackduck scan security risks

### ISSUE THREAD

**Issue by rajeshkatkarnice** [apache/airflow#14253]

**Comment by potiuk** [comment]
Hello @rajeshkatkarnice. Thanks for reporting those, but this is not the way security issues should be reported, this report is not actionable.

One comment - I know BlackDuck first hand and I know that it often reports issues which are not existing. We have no BlackDuck licences so we cannot reproduce your results, so it is difficult to act on your report.

First of all, can you please re-run your checks on 1.10.14 if you want to use 1.10 line. There were many changes and updates since 1.10.10 and it is likely at least some of those problems have been fixed in subsequent versions. We are not releasing patches to earlier versions. If there are any critical security fixes we will release them in 1.10.15, 1.10.16 - there won't ever be 1.10.10.1 release for exmple. So I will kindly ask you to re-run your Black Duck scan on 1.10.14 if you want some action to be taken.

Secondly - if you do, suspect any problems with security, you report it via email to security@apache.org - not via public issue. This is detailed in https://www.apache.org/security/ and the rules there are rather clear:

```
Please send one plain-text email for each vulnerability you are reporting. We may ask you to resubmit your report if you send it as an image, movie, HTML, or PDF attachment when it could just as easily be described with plain text.
```

So you should break it down into separate issues and sent it there.

This list is private and this is the "responsible disclosure" process that you should follow. You should never discuss even suspected security issues in public issues. It also allows to properly process the issues with assigning CVEs when the issue is triaged and found a real issue. 

So may I ask you - kindly - rerun your scans on latest 1.10 version (possibly also on latest 2.0 version) and follow the process of reporting the issues this way. Just for the sake of easy discovery I will edit the issue now and close it.

**Comment by potiuk** [comment]
Also - those rules are explained when you report issues - when you create "New Issue" there is quite clear "Report security issue" link that leads you to the ASF security paage.

**Comment by rajeshkatkarnice** [comment]
@potiuk we already have communicated these issue over mail to Apache Security Team before raising these issues.

We got below reply from Mark(mjc@gsuite.cloud.apache.org )

//
Hi, 
 
Outdated dependencies are not always security issues.  A project would only be affected if a dependency was used in such a way that the affected underlying code is used and the vulnerabilities were exposed.  We typically get reports sent to us from scanning tools that looks at dependencies out of context on how they are actually used in the projects.  As such we reject these reports and suggest you either a) show how the product is affected by the dependency vulnerabilities, or b) simply mention this as a normal bug report to that project.  Since dependency vulnerabilities are quite public, there is no need to use this private reporting mechanism for them.
 
Regards, Mark

//

**Comment by potiuk** [comment]
The answer you get is becauyse you lilely sent them the full report and it was not following this:
```
Please send one plain-text email for each vulnerability you are reporting.
```

As the message from security team says - just a report produced by any tool is out of context and it does not mean that the code is affected. Those reports are often misleading, so it would be great to make use of the report, but:

1) Rerunning it on 1.10.14 might limit the number of reported vulnerabilities

2) Submitting it in one big issue is not really helpful. If you would like to report it to us, that is great, but Ideally it should be itemised into separate issues per vulnerability and pre-screened by you. This is a community driven project, so there is no single person that can review the report and decide what to do with every single problem there - but since you have the tool and experience wiht it, you can actually help the community and pre-screen those issues. You can help by reviewing and decide yourself if this is a real issue affecting the UI or not and only report it afterwards:

* If you think it is critical security issue - report it separately (one -per item) to security@apache.org in private
* If you think it's not critical, you might  check if it is already fixed in 2.0 (should be easy to check by looking at the package versions) - we are not likely to fix non-critical issues in 1.10 line.
* if you think that  there is an important reasons for us to update, then please submit it is as a separate issue each (explaining why you think it is worth to fix in 1.10 line). 

Generally itemizing the issues should be helpful also for you to understand if those issues are worth reporting at all.
